l10n_br_base/models/res_partner_bank.py

```diff
@@ -3,23 +3,63 @@
 # Copyright (C) 2012 Raphaël Valyi (Akretion)
 # License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
 
-from odoo import fields, models
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+BANK_ACCOUNT_TYPE = [
+    ('01', _('Conta corrente individual')),
+    ('02', _('Conta poupança individual')),
+    ('03', _('Conta depósito judicial/Depósito em consignação individual')),
+    ('11', _('Conta corrente conjunta')),
+    ('12', _('Conta poupança conjunta')),
+    ('13', _('Conta depósito judicial/Depósito em consignação conjunta')),
+]
 
 
 class ResPartnerBank(models.Model):
     """ Adiciona campos necessários para o cadastramentos de contas
     bancárias no Brasil."""
+    _inherit = 'res.partner.bank'
 
-    _inherit = "res.partner.bank"
+    bank_account_type = fields.Selection(
+        selection=BANK_ACCOUNT_TYPE,
+        string='Bank Account Type',
+        default='01',
+    )
 
-    acc_number = fields.Char(string="Account Number", size=64, required=False)
+    acc_number = fields.Char(
+        string='Account Number',
+        size=64,
+        required=False,
+    )
 
-    acc_number_dig = fields.Char(string="Account Digit", size=8)
+    acc_number_dig = fields.Char(
+        string='Account Digit',
+        size=8,
+    )
 
-    bra_number = fields.Char(string="Bank Branch", size=8)
+    bra_number = fields.Char(
+        string='Bank Branch',
+        size=8,
```

Comment:
@renatonlima Na refatoração está com 4 digitos porém na interface está com 8 dígitos

![image](https://user-images.githubusercontent.com/52328080/86068115-df9b5000-ba4c-11ea-9283-00c620db30b5.png)

Reply:
@jpcarassato O campo deve ficar com o tamanho original, porque pode ser cadastrado uma conta de um banco de outro país por exemplo, por isso que temos a validação do campo que se for um banco do brasil deve ter 4 dígitos.

Reply:
> @jpcarassato O campo deve ficar com o tamanho original, porque pode ser cadastrado uma conta de um banco de outro país por exemplo, por isso que temos a validação do campo que se for um banco do brasil deve ter 4 dígitos.

OK @renatonlima

Reply:
@jpcarassato sobre as restricoes que vao dentro do banco de dados, a gente tem que tomar cuidado para nao ferrar o uso em empresas multi-nacionais (as que vao investir $$ para ter Odoo legal) ou empresas que tem alguma atividade fora do Brasil. Nos modulos super centrais da localizacao na v12, tivemos esse cuidado no design do modelo de dados, com esse tipo de examplo.